src/main/java/frc/robot/subsystems/NewVision/NewVisionIO.java

```diff
@@ -0,0 +1,137 @@
+            for (int i = 0; i <4; i++) {
+                detectedCornersX[i] = target.getDetectedCorners().get(i).x;
+                detectedCornersY[i] = target.getDetectedCorners().get(i).y;
+                minAreaRectCornersX[i] = target.getMinAreaRectCorners().get(i).x;
+                minAreaRectCornersY[i] = target.getMinAreaRectCorners().get(i).y;
+            }
+            table.put("detected corners x " + name, detectedCornersX);
+            table.put("detected corners y " + name, detectedCornersY);
+            table.put("min area rect corners x " + name, minAreaRectCornersX);
+            table.put("min area rect corners Y " + name, minAreaRectCornersY);
+        }
+
+        public static void logTransform3d(Transform3d transform3d, LogTable table, String name) {
+            double rotation[] = new double[4];
+            rotation[0] = transform3d.getRotation().getQuaternion().getW();
+            rotation[1] = transform3d.getRotation().getQuaternion().getX();
+            rotation[2] = transform3d.getRotation().getQuaternion().getY();
+            rotation[3] = transform3d.getRotation().getQuaternion().getZ();
+            table.put("rotation " + name, rotation);
+
+            double translation[] = new double[3];
+            translation[0] = transform3d.getTranslation().getX();
+            translation[1] = transform3d.getTranslation().getY();
+            translation[2] = transform3d.getTranslation().getZ();
+            table.put("translation " + name, translation);
+        }
+
+        public Transform3d getLoggedTransform3d(double[] translation, double[] rotation) {
+            Transform3d transform3d = new Transform3d(
+                new Translation3d(
+                    translation[0], 
+                    translation[1], 
+                    translation[2]),
+                new Rotation3d(
+                    new Quaternion(
+                        rotation[0], 
+                        rotation[1], 
+                        rotation[2], 
+                        rotation[3]
+                    )
+                )
+            );
+            return transform3d;
+        }
+
+        public void getLoggedPhotonTrackedTarget(PhotonTrackedTarget target, LogTable table, String name) {
+            double[] translation = table.getDoubleArray("translation " + name, new double[3]);
+            double[] rotation = table.getDoubleArray("rotation " + name, new double[4]);
+            double[] altTranslation = table.getDoubleArray("translation alt " + name, new double[3]);
+            double[] altRotation = table.getDoubleArray("rotation alt " + name, new double[4]);
+
+            List<TargetCorner> detectedCorners = new ArrayList<>();
+            List<TargetCorner> minAreaRectCorners = new ArrayList<>();
+
+            for (int i = 0; i <4; i++) {
+                detectedCorners.add(new TargetCorner(detectedCornersX[i], detectedCornersY[i]));
```

Comment:
calling to the static variable here wont get you the logged value. Instead use `table.getDoubleArray("detected corners x", new double[4])` or something similar